### DISCUSSION TITLE
Modeling African American (admixed) populations

### DISCUSSION THREAD

**Discussion by lakishadavid** [tskit-dev/msprime#2015] [Q&A] [answered]
How do I use msprime to model African American populations?

I followed the msprime tutorial and reached the demographic model example for Admixture and the American Admixture model based on [Browning et al. 2011](https://doi.org/10.1371/journal.pgen.1007385). The admixture and simulation part of the code is as follows:

```
demography.add_admixture(
    12,
    derived="ADMIX",
    ancestral=["AFR", "EUR", "EAS"],
    proportions=[1 / 6, 2 / 6, 3 / 6],
)

demography.add_population_split(920, derived=["EUR", "EAS"], ancestral="OOA")

demography.add_symmetric_migration_rate_change(
    time=920, populations=["AFR", "OOA"], rate=15e-5
)

demography.add_population_split(2040, derived=["OOA", "AFR"], ancestral="AMH")
demography.add_population_split(5920, derived=["AMH"], ancestral="ANC")
demography.sort_events()
demography

ts = msprime.sim_ancestry(
    samples={
        "AFR": 100, 
        "ADMIX": 100},
    model="dtwf", 
    demography=demography,
    random_seed=12
    )
ts
``` 
From my understanding, this creates a population where 1/6 of the individuals have African ancestry, 1/3 have European ancestry, and 1/2 have East Asian ancestry. However, I need to model an admixed population where each individual has admixed ancestry. The individuals would have, for example, 84.9% African, 14% European, and 1.1% Native American (on average). Then, I would sample from this African American population and the African population for further analysis. Is there example code that shows how to do this?

**Comment by andrewkern** [comment]
Hi @lakishadavid - if you are interested in simulating from the Browning et al model, let me point you over to our [stdpopsim project](https://popsim-consortium.github.io/stdpopsim-docs/stable/introduction.html), in case you don't yet know about it.

The Browning et al model is [actually already implemented](https://popsim-consortium.github.io/stdpopsim-docs/stable/catalog.html#sec_catalog_homsap_models_americanadmixture_4b11) in stdpopsim as one of the human models, so could provide a quick and well tested route towards what you want.

Sorry to not answer your question though!

**Reply by lakishadavid** [reply]
Thanks, @andrewkern . stdpopsim would be great if I could find a model for African Americans. The ADMIX population from AmericanAdmixture_4B11, I don't get how this works out to represent the varying levels of admixture in African American, European American, and Native American groups today based on works like [The Great Migration and African-American Genomic Diversity](https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1006059). ...which I guess is fine if you're stratifying segments by continental ancestry before doing IBD analysis anyway.

**Reply by molpopgen** [reply]
Using a demes.Graph written as a YAML file will be simpler than using the low-level msprime API; https://tskit.dev/msprime/docs/stable/api.html#msprime.Demography.from_demes.

The link will take you to the demes docs via the msprime docs.

**Comment by jeromekelleher** [comment, marked as answer]
Hi @lakishadavid :wave:

There's no problem having admixed populations deriving from other admixed populations, you should be able to set them up in the same way. You can sample from any population that you want, using the ``samples`` argument.

The tools in Demes are much better for working with complex demographies though, as @molpopgen says. The [demes](https://popsim-consortium.github.io/demes-docs/main/introduction.html) Python package is a good place to startl.

**Reply by lakishadavid** [reply]
@jeromekelleher , do you have an estimation of when the pedigree simulator will be pushed out? Depending on when we could use it and its capabilities, I may wait to use that for the project I'm working on. I would like to simulate a case where African American and African populations in a sample have MRCAs from 4 to 9 generations ago. From this, I would look at stats and topology and compare it to real data. Would the pedigree simulator be able to do this simulation or should I continue with other methods?

**Reply by jeromekelleher** [reply]
It's basically ready to use @lakishadavid, it just needs some final documentation polishing and to be released. The best thing to do woul be to have a look at the [documentation](https://tskit.dev/msprime/docs/latest/ancestry.html#fixed-pedigree) for the fixed pedigree and see if it looks like it does what you need. If you see any problems, please do let us know also!
